core/server/src/test/java/alluxio/master/file/PermissionCheckerTest.java

```diff
@@ -194,6 +194,14 @@ public void before() throws Exception {
     mPermissionChecker = new PermissionChecker(sTree);
   }
 
+  // Helper function to create a path and set the permission to what specified in option.
+  private static void createAndSetPermission(String path, CreateFileOptions option)
+      throws Exception {
+    InodeTree.CreatePathResult result = sTree.createPath(new AlluxioURI(path), option);
+    result.getCreated().get(result.getCreated().size() - 1)
```

Comment:
I should be more specific here. right now we move the logic to apply umask on newly created file or dir on `InodeTree.createPath` which is responsible for creating new inode. That's why by calling `sTree.createPath()`, one could not create some files/dirs with weird permission bits directly (as we are doing here, the tests requires to create a dir with permission bit 0157, in the normal code path, 0157 will be umasked, and the way to create such a file is to update permission later where no umask will be applied )